validator-cli/src/ArbToEth/validator.test.ts

```diff
@@ -0,0 +1,220 @@
+import { ethers } from "ethers";
+import { challengeAndResolveClaim } from "./validator";
+import { BotEvents } from "../utils/botEvents";
+
+describe("validator", () => {
+  let veaOutbox: any;
+  let veaInbox: any;
+  let veaInboxProvider: any;
+  let veaOutboxProvider: any;
+  let emitter: any;
+  let mockGetClaim: any;
+  let mockClaim: any;
+  let mockGetClaimState: any;
+  let mockGetBlockFinality: any;
+
+  beforeEach(() => {
+    veaInbox = {
+      snapshots: jest.fn(),
+      provider: {
+        getBlock: jest.fn(),
+      },
+    };
+    veaOutbox = {
+      claimHashes: jest.fn(),
+      queryFilter: jest.fn(),
+      provider: {
+        getBlock: jest.fn(),
+      },
+    };
+    emitter = {
+      emit: jest.fn(),
+    };
+    mockClaim = {
+      stateRoot: "0x1234",
+      claimer: "0xFa00D29d378EDC57AA1006946F0fc6230a5E3288",
+      timestampClaimed: 1234,
+      timestampVerification: 0,
+      blocknumberVerification: 0,
+      honest: 0,
+      challenger: ethers.ZeroAddress,
+    };
+    mockGetClaim = jest.fn();
+    mockGetBlockFinality = jest.fn().mockResolvedValue([{ number: 0 }, { number: 0, timestamp: 100 }, false]);
+  });
+  describe("challengeAndResolveClaim", () => {
+    it("should return null if no claim is made", async () => {
+      const transactionHandler = null;
+      mockGetClaim = jest.fn().mockReturnValue(null);
+      const result = await challengeAndResolveClaim(
```

Comment:
If the function accepts an object instead of positional arguments, you can create a default object with all the mocked values: assertions will only require to spread the default object and place the few values that are relevant for the test case.